resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/BasicTestsSuite.scala

```diff
@@ -107,6 +107,24 @@ private[spark] trait BasicTestsSuite { k8sSuite: KubernetesSuite =>
       runSparkRemoteCheckAndVerifyCompletion(appArgs = Array(REMOTE_PAGE_RANK_FILE_NAME))
     }
   }
+
+  test("SPARK-36321: Do not fail application in kubernetes if name is too long", k8sTestTag) {
+    def executorPodCheck(executorPod: Pod): Unit = {
+      assert(executorPod.getMetadata.getName.startsWith("abbbbb"))
```

Comment:
please go up to "-" in this `startsWith` and then cut out the part from this `executorPod.getMetadata.getName` where the prefix and the unique ID is and in the second assert use that string in the `startsWith`